src/lib/io/sol-gpio-impl-zephyr.c

```diff
@@ -33,35 +34,37 @@
 
 SOL_LOG_INTERNAL_DECLARE_STATIC(_log_domain, "gpio");
 
-static atomic_val_t int_flag = 0;
-static uint32_t irq_pins;
-
-static struct sol_ptr_vector opened_pins = SOL_PTR_VECTOR_INIT;
-
-/* Zephyr structure used as a "bank of gpio pins" */
-static struct device *device;
+struct sol_gpio_port {
+    struct device *device;
+    struct gpio_callback cb;
+    struct sol_ptr_vector opened_pins;
+    uint32_t irq_pins;
+    atomic_val_t int_flag;
+};
 
 struct sol_gpio {
     uint32_t pin;
-    bool active_low;
+    struct sol_gpio_port *port;
     void (*cb)(void *data, struct sol_gpio *gpio, bool value);
     const void *cb_data;
+    bool active_low;
 };
 
 static void
 sol_gpio_interrupt_process(void *data)
 {
+    struct sol_gpio_port *port = data;
     struct sol_gpio *g;
     uint32_t irq;
     uint16_t i;
 
-    if (!atomic_cas(&int_flag, 1, 0))
+    if (!atomic_cas(&port->int_flag, 1, 0))
         return;
 
-    irq = irq_pins;
-    irq_pins = 0;
+    irq = port->irq_pins;
+    port->irq_pins = 0;
 
-    SOL_PTR_VECTOR_FOREACH_IDX (&opened_pins, g, i) {
+    SOL_PTR_VECTOR_FOREACH_IDX (&port->opened_pins, g, i) {
         if (irq & (1U << g->pin)) {
```

Comment:
On Tue, 26 Apr 2016 15:15:43 -0700, Gustavo Lima Chaves wrote:
> >  
> > -    SOL_PTR_VECTOR_FOREACH_IDX (&opened_pins, g, i) {
> > +    SOL_PTR_VECTOR_FOREACH_IDX (&port->opened_pins, g, i) {
> >          if (irq & (1U << g->pin)) {
> 
> You could use ``BIT()`` on those as well.

Ugh, that was already like that and I didn't notice it.